korge/src/commonMain/kotlin/com/soywiz/korge/view/camera/Camera.kt

```diff
@@ -0,0 +1,173 @@
+            this.angle = other.angle
+            this.anchorX = other.anchorX
+            this.anchorY = other.anchorY
+        }
+        return this
+    }
+
+    suspend fun tweenTo(other: Camera, time: TimeSpan, easing: Easing = Easing.LINEAR) {
+        container?.tweenTo(other, time = time, easing = easing)
+        withoutUpdate { setTo(other) }
+    }
+
+    fun follow(view: View, threshold: Double = 0.0) {
+        val inside = container?.content?.let { view.hasAncestor(it) } ?: false
+        if (!inside) throw IllegalStateException("Can't follow view that is not in the content of CameraContainer")
+        cancelFollowing?.cancel()
+        cancelFollowing = view.addUpdater {
+            val camera = this@Camera
+            val cxcenter = camera.x + camera.width * camera.anchorX
+            val cycenter = camera.y + camera.height * camera.anchorY
+            val vxcenter = x + width / 2
+            val vycenter = y + height / 2
+            val dx = cxcenter - vxcenter
+            val dy = cycenter - vycenter
+            val adx = abs(dx) - threshold
+            val ady = abs(dy) - threshold
+            if (adx > 0 || ady > 0) {
+                camera.xy(
+                    camera.x - dx.sign * adx,
+                    camera.y - dy.sign * ady
+                )
+            }
+        }
+    }
+
+    fun unfollow() {
+        cancelFollowing?.cancel()
+    }
+
+    fun copy(
+        x: Double = this.x,
+        y: Double = this.y,
+        width: Double = this.width,
+        height: Double = this.height,
+        zoom: Double = this.zoom,
+        angle: Angle = this.angle,
+        anchorX: Double = this.anchorX,
+        anchorY: Double = this.anchorY
+    ) = Camera(
+        x = x,
+        y = y,
+        width = width,
+        height = height,
+        zoom = zoom,
+        angle = angle,
+        anchorX = anchorX,
+        anchorY = anchorY
+    )
+
+    override fun setToInterpolated(ratio: Double, l: Camera, r: Camera) = setTo(
+        x = ratio.interpolate(l.x, r.x),
+        y = ratio.interpolate(l.y, r.y),
+        width = ratio.interpolate(l.width, r.width),
+        height = ratio.interpolate(l.height, r.height),
+        zoom = ratio.interpolate(l.zoom, r.zoom),
+        angle = ratio.interpolate(l.angle, r.angle),
+        anchorX = ratio.interpolate(l.anchorX, r.anchorX),
+        anchorY = ratio.interpolate(l.anchorY, r.anchorY)
+    )
+
+    override fun interpolateWith(ratio: Double, other: Camera) = Camera().setToInterpolated(ratio, this, other)
+}
+
+fun Camera.xy(x: Double, y: Double): Camera {
+    return setTo(x = x, y = y)
+}
+
+fun Camera.size(width: Double, height: Double): Camera {
+    return setTo(width = width, height = height)
+}
+
+fun Camera.anchor(anchorX: Double, anchorY: Double): Camera {
+    return setTo(anchorX = anchorX, anchorY = anchorY)
+}
+
+suspend fun Camera.moveTo(x: Double, y: Double, time: TimeSpan, easing: Easing = Easing.LINEAR) {
```

Comment:
I don't know because returning camera would be useful for chained calls and I don't think these suspend functions with the effect of code postponing should be used in chained calls (a user could confuse them with the view dsl functions).